stdlib/SparseArrays/test/sparse.jl

```diff
@@ -384,6 +384,52 @@ end
     end
 end
 
+@testset "multiplication of sparse matrix and triangular matrix" begin
+    function _sparse_test_matrix(n, T)
+        if T == Int
+            return sparse(rand(0:4, n, n))
+        else
+            return sprandn(T, n, n, 0.6)
+        end
+    end
+    function _triangular_test_matrix(n, TA, T)
+        if T == Int
+            return TA(rand(0:9, n, n))
+        else
+            return TA(randn(T, n, n))
+        end
+    end
+
+    n = 5
+    for T1 in (Int, Float64, ComplexF32)
+        S = _sparse_test_matrix(n, T1)
+        MS = Matrix(S)
+        for T2 in (Int, Float64, ComplexF32)
+            for TM in (LowerTriangular, UnitLowerTriangular, UpperTriangular, UnitLowerTriangular)
+                T = _triangular_test_matrix(n, TM, T2)
+                MT = Matrix(T)
+                @test T * S ≈ MT * MS
+                @test isa(T * S, DenseMatrix)
+                @test T' * S ≈ MT' * MS
+                @test transpose(T) * S ≈ transpose(MT) * MS
+                @test T * S' ≈ MT * MS'
+                @test T * transpose(S) ≈ MT * transpose(MS)
+                @test T' * transpose(S) ≈ MT' * transpose(MS)
+                @test transpose(T) * S' ≈ transpose(MT) * S'
```

Comment:
Just a suggestion:
```julia
@test isa(T * S, DenseMatrix)
for transT in (identity, adjoint, transpose), transS in (identity, adjoint, transpose)
    @test transT(T) * transS(S) ≈ transT(MT) * transS(MS)
end
```
That should cover all cases, if I'm not mistaken.

Reply:
Oh, and if you add one line with `T` and `S` interchanged, then that should cover the second block...?

Reply:
You may know already, but just to be explicit: what I like about this pattern also is that it helps reducing copy-paste errors.